src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/Byte/ByteFrozenSet.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace System.Collections.Frozen
+{
+    /// <summary>Provides a frozen set to use when the values are <see cref="byte"/>s and the default comparer is used.</summary>
+    internal sealed partial class ByteFrozenSet : FrozenSetInternalBase<byte, ByteFrozenSet.GSW>
+    {
+        private readonly BitVector256 _values;
+        private readonly int _count;
+        private byte[]? _items;
+
+        internal ByteFrozenSet(ReadOnlySpan<byte> values) : base(EqualityComparer<byte>.Default)
+        {
+            Debug.Assert(!values.IsEmpty);
+
+            foreach (byte b in values)
+            {
+                _values.Set(b);
+            }
+
+            _count = _values.Count();
+        }
+
+        internal ByteFrozenSet(IEnumerable<byte> values) : base(EqualityComparer<byte>.Default)
+        {
+            foreach (byte b in values)
+            {
+                _values.Set(b);
+            }
+
+            _count = _values.Count();
+
+            Debug.Assert(_count > 0);
+        }
+
+        /// <inheritdoc />
+        private protected override byte[] ItemsCore => _items ??= _values.GetByteValues();
```

Comment:
Nit: you can use a semi-auto properties here 🙂
```suggestion
        [field: MaybeNull]
        private protected override byte[] ItemsCore => field ??= _values.GetByteValues();
```

Same in the other type.